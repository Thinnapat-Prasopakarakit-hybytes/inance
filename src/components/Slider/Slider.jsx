import React from "react";
import "./Slider.scss";
import { Link } from "react-router-dom";
import img from "../../assets/images/slider-img.png";
import { useIntl } from "react-intl";

const Slider = () => {
  const { messages } = useIntl();
  return (
    <section className="slider_section">
      <div className="container">
        <div className="row">
          <div className="col-md-6">
            <div className="detail-box">
              <h1>
                {messages.slider.firstLine} <br />
                {messages.slider.secondLine} <br />
                {messages.slider.thirdLine}
              </h1>
              <p>{messages.slider.description}</p>
              <Link to="/contact"> {messages.slider.contact} </Link>
            </div>
          </div>
          <div className="col-md-6">
            <div className="img-box">
              <img src={img} alt="slider img" />
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Slider;
