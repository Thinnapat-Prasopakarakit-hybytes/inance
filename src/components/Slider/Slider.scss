@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.slider_section {
  flex: 1;
  display: flex;
  align-items: center;
  background-size: cover;
  background-position: bottom;
  padding-top: 0;
  @include desktop {
    padding-top: 45px;
  }

  .row {
    align-items: center;
  }

  .detail-box {
    padding-bottom: 90px;
    @include tablet {
      ppadding-bottom: 45px;
    }

    h1 {
      font-weight: bold;
      margin-bottom: 25px;
      color: $primary;
    }

    p {
      color: #444;
      font-size: 15px;
      line-height: 22.5px;
    }

    a {
      @include hero_btn($primary2, $white, 10px, 45px, 5px);
      margin-top: 15px;
      text-decoration: none !important;

      &:hover,
      &:focus,
      &:active {
        text-decoration: none !important;
      }
    }
  }

  .img-box {
    img {
      width: 100%;
    }
  }
}
