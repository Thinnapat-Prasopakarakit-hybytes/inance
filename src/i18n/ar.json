{
  "header": {
    "phone": "اتصل : +01 123455678990",
    "email": "البريد الإلكتروني : demo@gmail.com"
  },
  "nav": {
    "home": "الرئيسية",
    "about": "من نحن",
    "services": "خدماتنا",
    "contact": "اتصل بنا"
  },
  "feature": {
    "repair": "إصلاح",
    "improve": "تحسين",
    "maintain": "صيانة"
  },
  "slider": {
    "firstLine": "إصلاح و",
    "secondLine": "صيانة",
    "thirdLine": "خدمات",
    "description": "لوريم إيبسوم دولور سيت أميت، كونسيكتيتور أديباسينغ إليت. نيمو إنيم، فيريتات أوديو، ليبريتاس لوريم إيبسوم إنتيجريت. مينيما كوي أوت أتشينت إينيم أوديو أكوربوز، سوسيب تينيت أورناريه إلتراتو نيمو ريسيتات. إني أمبيد كوي أتالوغ إيرابيل، إيس أونجتوت أوديو لوريم. أوديا فان إينيس ديوس غريلانديسكو فوريس، أرتيموس كانايور.",
    "contact": "اتصل بنا"
  },
  "about": {
    "title": "من نحن",
    "readMore": "اقرأ المزيد",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص."
  },
  "professional": {
    "title": "نحن نقدم خدمات",
    "subtitle": "منزلية احترافية",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص.",
    "readMore": "اقرأ المزيد"
  },
  "services": {
    "title": "خدماتنا",
    "maintenance": "صيانة",
    "electrical": "كهرباء",
    "plumbing": "سباكة",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ",
    "viewMore": "عرض المزيد"
  },
  "contact": {
    "title": "اتصل بنا",
    "name": "الاسم",
    "phone": "رقم الهاتف",
    "email": "البريد الإلكتروني",
    "message": "الرسالة",
    "send": "إرسال"
  },
  "client": {
    "title": "ماذا يقول عملاؤنا",
    "name": "جورش موريك",
    "testimonial": "يقوم بتقسيم النص إلى أجزاء حسب الضرورة، مما يجعله أول مولد حقيقي على الإنترنت. يستخدم قاموسًا يحتوي على أكثر من 200 كلمة لاتينية، مع عدد قليل من تراكيب الجمل النموذجية، لتوليد نص لوريم إيبسوم."
  },
  "info": {
    "getInTouch": "تواصل معنا",
    "text": "لوريم إيبسوم هو نص تجريدي بسيط",
    "followUs": "تابعنا"
  },
  "footer": {
    "copyright": "جميع الحقوق محفوظة لـ",
    "company": "قوالب HTML المجانية"
  },
  "notFound": {
    "text": "الصفحة غير موجودة",
    "description": "عذراً! الصفحة التي تبحث عنها غير موجودة",
    "link": "العودة إلى الصفحة الرئيسية"
  }
}
