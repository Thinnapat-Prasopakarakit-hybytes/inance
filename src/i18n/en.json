{
  "header": {
    "phone": "Call : +01 123455678990",
    "email": "Email : demo@gmail.com"
  },
  "nav": {
    "home": "Home",
    "about": "About",
    "services": "Services",
    "contact": "Contact Us"
  },
  "feature": {
    "repair": "Repair",
    "improve": "Improve",
    "maintain": "Maintain"
  },
  "slider": {
    "firstLine": "Repair and",
    "secondLine": "Maintenance",
    "thirdLine": "Services",
    "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui harum voluptatem adipisci. Quos molestiae saepe dicta nobis pariatur, tempora iusto, ad possimus soluta hic praesentium mollitia consequatur beatae, aspernatur culpa.",
    "contact": "Contact Us"
  },
  "about": {
    "title": "About us",
    "readMore": "Read More",
    "description": "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable."
  },
  "professional": {
    "title": "We Provide Professional",
    "subtitle": "Home Services.",
    "description": "randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All randomised words which don't look even slightly",
    "readMore": "Read More"
  },
  "services": {
    "title": "Our Services",
    "maintenance": "Maintenance",
    "electrical": "Electrical",
    "plumbing": "Plumbing",
    "description": "when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal",
    "viewMore": "View More"
  },
  "contact": {
    "title": "Contact Us",
    "name": "Name",
    "phone": "Phone Number",
    "email": "Email",
    "message": "Message",
    "send": "SEND"
  },
  "client": {
    "title": "What Our Clients Say",
    "name": "Jorch morik",
    "testimonial": "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum"
  },
  "info": {
    "getInTouch": "Get In Touch",
    "text": "Lorem Ipsum is simply dummy text",
    "followUs": "Follow Us"
  },
  "footer": {
    "copyright": "All Rights Reserved By",
    "company": "Free Html Templates"
  },
  "notFound": {
    "text": "Page not found",
    "description": "Oops! The page you are looking for does not exist",
    "link": "Back to Home"
  }
}
