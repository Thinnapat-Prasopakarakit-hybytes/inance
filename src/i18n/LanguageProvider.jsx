import { useState, createContext, useEffect } from "react";
import { IntlProvider } from "react-intl";
import { useLocation, useParams } from "react-router-dom";
import en from "./en.json";
import ar from "./ar.json";
import { HelmetProvider } from "react-helmet-async";
import SEO from "../components/SEO";
import { logEvent, logLanguageChange } from "../analytics";

export const LanguageContext = createContext();

const messages = {
  en,
  ar,
};

const LanguageProvider = ({ children }) => {
  const { lang } = useParams();
  const { pathname } = useLocation();
  const [locale, setLocale] = useState(lang === "ar" ? "ar" : "en");

  useEffect(() => {
    if (lang && (lang === "en" || lang === "ar")) {
      localStorage.setItem("lang", lang);
      setLocale(lang);
      document.documentElement.dir = lang === "ar" ? "rtl" : "ltr";
      document.documentElement.lang = lang;
    }
  }, [lang]);

  const handleLanguageChange = (newLocale) => {
    if (locale !== newLocale) {
      logEvent("Language", "Change", newLocale);
      const pathWithoutLang = pathname.replace(/^\/(en|ar)/, "");
      const newPath =
        pathWithoutLang === "" || pathWithoutLang === "/"
          ? `/${newLocale}`
          : `/${newLocale}${pathWithoutLang}`;

      window.location.href = newPath;
    }
  };

  return (
    <HelmetProvider>
      <LanguageContext.Provider value={{ locale, handleLanguageChange }}>
        <IntlProvider
          messages={messages[locale]}
          locale={locale}
          defaultLocale="en"
        >
          <SEO lang={locale} />
          {children}
        </IntlProvider>
      </LanguageContext.Provider>
    </HelmetProvider>
  );
};

export default LanguageProvider;
