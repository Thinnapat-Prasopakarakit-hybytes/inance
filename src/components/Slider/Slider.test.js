import React from "react";
import { render, screen } from "@testing-library/react";
import Slider from "./Slider";
import { BrowserRouter } from "react-router-dom";
import { LanguageContext } from "../../i18n/LanguageProvider";
import { IntlProvider } from "react-intl";
import en from "../../i18n/en.json";
import ar from "../../i18n/ar.json";

const messages = {
  en,
  ar,
};

const renderSlider = (locale = "en") => {
  return render(
    <BrowserRouter>
      <LanguageContext.Provider value={{ locale }}>
        <IntlProvider messages={messages[locale]} locale={locale}>
          <Slider />
        </IntlProvider>
      </LanguageContext.Provider>
    </BrowserRouter>
  );
};

describe("Slider Component", () => {
  it("should render the main heading", () => {
    renderSlider();
    const heading = screen.getByText(/Repair and/i);
    expect(heading).toBeInTheDocument();
  });

  it("should render the description text", () => {
    renderSlider();
    const description = screen.getByText(/Lorem ipsum dolor sit amet/i);
    expect(description).toBeInTheDocument();
  });

  it("should render the contact us link", () => {
    renderSlider();
    const contactLink = screen.getByText(/Contact Us/i);
    expect(contactLink).toBeInTheDocument();
    expect(contactLink).toHaveAttribute("href", "/en/contact");
  });

  it("should render the slider image", () => {
    renderSlider();
    const image = screen.getByRole("img");
    expect(image).toHaveAttribute("src", "mock-slider-img");
    expect(image).toHaveAttribute("alt", "slider img");
  });

  it("should match snapshot", () => {
    const { asFragment } = renderSlider();
    expect(asFragment()).toMatchSnapshot();
  });
});
