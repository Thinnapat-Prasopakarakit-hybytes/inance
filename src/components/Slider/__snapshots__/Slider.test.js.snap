// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="slider_section"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <div
            class="detail-box"
          >
            <h1>
              Repair and 
              <br />
              Maintenance 
              <br />
              Services
            </h1>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui harum voluptatem adipisci. Quos molestiae saepe dicta nobis pariatur, tempora iusto, ad possimus soluta hic praesentium mollitia consequatur beatae, aspernatur culpa.
            </p>
            <a
              href="/en/contact"
            >
               Contact Us 
            </a>
          </div>
        </div>
        <div
          class="col-md-6"
        >
          <div
            class="img-box"
          >
            <img
              alt="slider img"
              src="mock-slider-img"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
